python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -247,12 +247,12 @@ def __init__(
                 self._selected_asset_check_keys = selected_asset_check_keys
 
         self._check_specs_by_output_name = {
-            name: spec
-            for name, spec in (check_specs_by_output_name or {}).items()
-            if spec.key in self._selected_asset_check_keys
+            name: spec for name, spec in (check_specs_by_output_name or {}).items()
```

Comment:
these code paths feel confusing because the names are confusing. check_specs_by_output_name makes sense here. Not that I feel like this PR is the right place to fix it, but it feels like we should just call things in terms of asset_key instead of node, and we should also be explicit around when things are subsetted.